### PR TITLE
Rename the Unit Testing side Json

### PR DESCRIPTION
**Pull request by Cosifne**
Address this [issue](https://github.com/dotnet/roslyn/issues/42582) for unit testing.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs [hunk 1]
```diff
@@ -22,9 +22,9 @@ namespace Microsoft.CodeAnalysis.Remote
         internal const string Prefix = "roslyn";
         internal const string IntelliCodeServiceName = "pythia";
         internal const string RazorServiceName = "razorLanguageService";
-        internal const string UnitTestingAnalysisServiceName = "remoteAnalysisService";
-        internal const string LiveUnitTestingServiceName = "remoteBuildService";
-        internal const string TestSourceLocationLookupServiceName = "remoteTestSourceLocationLookupService";
+        internal const string UnitTestingAnalysisServiceName = "unitTestAnalysis";
```
Comment (Cosifne) [inline src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs:25]: @shyamnamboodiripad for checking do these names make sense

Comment (tmat) [reply src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs:25]: Can you make these start with capital letter? (e.g. `UnitTestAnalysis`)

Comment (tmat) [reply src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs:25]: I updated https://github.com/dotnet/roslyn/issues/42582 - previously I wrote "CamelCasing" but I actually meant "PascalCasing". Sorry about the confusion.

Comment (tmat) [reply src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs:25]: Other non-Roslyn VS services use PascalCasing, so to be consistent.

File: src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs [hunk 2]
```diff
@@ -22,9 +22,9 @@ namespace Microsoft.CodeAnalysis.Remote
...
+        internal const string LiveUnitTestingServiceName = "liveUnitTestIncrementalBuild";
+        internal const string TestSourceLocationLookupServiceName = "unitTestSourceLookup";
```
Comment (shyamnamboodiripad) [inline src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs:27]: I'd prefer the following names -

```
UnitTestingAnalysis
UnitTestingSourceLookup
LiveUnitTestingBuild
```

File: src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs [hunk 3]
```diff
@@ -22,9 +22,9 @@ namespace Microsoft.CodeAnalysis.Remote
         internal const string Prefix = "roslyn";
         internal const string IntelliCodeServiceName = "pythia";
         internal const string RazorServiceName = "razorLanguageService";
-        internal const string UnitTestingAnalysisServiceName = "unitTestAnalysis";
-        internal const string LiveUnitTestingServiceName = "liveUnitTestIncrementalBuild";
-        internal const string TestSourceLocationLookupServiceName = "unitTestSourceLookup";
+        internal const string UnitTestingAnalysisServiceName = "UnitTestingAnalysis";
+        internal const string LiveUnitTestingServiceName = "UnitTestingSourceLookup";
+        internal const string TestSourceLocationLookupServiceName = "LiveUnitTestingBuild";
```
Comment (shyamnamboodiripad) [inline src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs:27]: The names for the LUT and SourceLocation services are mixed up :)

Also consider renaming the consts as well as the enum members to match the strings

Comment (Cosifne) [reply src/Workspaces/Core/Portable/Remote/RemoteServiceName.cs:27]: 😥Ah I didn't notice that.

File: src/Workspaces/Core/Portable/Remote/WellKnownServiceHubService.cs [hunk 4]
```diff
@@ -21,7 +21,7 @@ internal enum WellKnownServiceHubService
 
         // owned by Unit Testing team:
         UnitTestingAnalysisService,
-        LiveUnitTestingService,
-        TestSourceLocationLookupService,
+        LiveUnitTestingBuildService,
```
Comment (shyamnamboodiripad) [inline src/Workspaces/Core/Portable/Remote/WellKnownServiceHubService.cs:24]: @tmat This service may go away at some point in 16.8. Should we worry about compat breaks when we remove the enum entry? Would it be better to move this to the end of the list for this reason?

**Pull request merged**